### ISSUE TITLE
UWF_FIELD_NOT_INITIALIZED_IN_CONSTRUCTOR: false positive with lambdas

### ISSUE THREAD

**Issue by Vampire** [spotbugs/spotbugs#949]
```java
private List<Restriction<? super M>> availableRestrictions;

protected void doSetAvailableRestrictions(Instance<Restriction<? super M>> availableRestrictions) {
    this.availableRestrictions = ...;
    logger.info("Got {} restriction{} injected",
            () -> this.availableRestrictions.size(),
            () -> this.availableRestrictions.size() == 1 ? "" : 's');
}
```

The rule complains about the two usages of the field that was not set in the constructor.
But it is set one line above, so this should not be reported.
